github/github_test.go

```diff
@@ -201,6 +201,9 @@ func testNewRequestAndDoFailure(t *testing.T, methodName string, client *Client,
 	client.BaseURL.Path = "/api-v3/"
 	client.rateLimits[0].Reset.Time = time.Now().Add(10 * time.Minute)
 	resp, err = f()
+	if bypass := resp.Request.Context().Value(bypassRateLimitCheck); bypass != nil {
+		return
```

Comment:
I'm concerned that we are not checking the value of `err` here... might there be a case where we get an error that will be  ignored here?

Reply:
This helper explicitly sets the client up to trigger errors, and at this stage a rate limit error. As is, the test would fail if we check for `err != nil` for `methodName == "RateLimits"` as the request would result in a HTTP 404, since no handler has been created. We can set one up on `TestRateLimits_coverage` but I think it makes the test less straightforward, and the nominal case is already covered by other functions.